lib/Ssl/SslServerFeature.cpp

```diff
@@ -301,247 +191,37 @@ asio::ssl::context SslServerFeature::createSslContext() const {
-    result.append(", SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION");
-  }
-#endif
-
-#ifdef SSL_OP_SINGLE_ECDH_USE
-  if (opts & SSL_OP_SINGLE_ECDH_USE) {
-    result.append(", SSL_OP_SINGLE_ECDH_USE");
-  }
-#endif
-
-#ifdef SSL_OP_SINGLE_DH_USE
-  if (opts & SSL_OP_SINGLE_DH_USE) {
-    result.append(", SSL_OP_SINGLE_DH_USE");
-  }
-#endif
-
-#ifdef SSL_OP_EPHEMERAL_RSA
-  if (opts & SSL_OP_EPHEMERAL_RSA) {
-    result.append(", SSL_OP_EPHEMERAL_RSA");
-  }
-#endif
-
-#ifdef SSL_OP_CIPHER_SERVER_PREFERENCE
-  if (opts & SSL_OP_CIPHER_SERVER_PREFERENCE) {
-    result.append(", SSL_OP_CIPHER_SERVER_PREFERENCE");
-  }
-#endif
-
-#ifdef SSL_OP_TLS_ROLLBACK_BUG
-  if (opts & SSL_OP_TLS_ROLLBACK_BUG) {
-    result.append(", SSL_OP_TLS_ROLLBACK_BUG");
-  }
-#endif
-
-#ifdef SSL_OP_NO_SSLv2
-  if (opts & SSL_OP_NO_SSLv2) {
-    result.append(", SSL_OP_NO_SSLv2");
-  }
-#endif
-
-#ifdef SSL_OP_NO_SSLv3
-  if (opts & SSL_OP_NO_SSLv3) {
-    result.append(", SSL_OP_NO_SSLv3");
-  }
-#endif
-
-#ifdef SSL_OP_NO_TLSv1
-  if (opts & SSL_OP_NO_TLSv1) {
-    result.append(", SSL_OP_NO_TLSv1");
-  }
-#endif
-
-#ifdef SSL_OP_NO_TLSv1_2
-  if (opts & SSL_OP_NO_TLSv1_2) {
-    result.append(", SSL_OP_NO_TLSv1_2");
-  }
-#endif
-
-#ifdef SSL_OP_NO_TLSv1_1
-  if (opts & SSL_OP_NO_TLSv1_1) {
-    result.append(", SSL_OP_NO_TLSv1_1");
-  }
-#endif
-
-#ifdef SSL_OP_NO_DTLSv1
-  if (opts & SSL_OP_NO_DTLSv1) {
-    result.append(", SSL_OP_NO_DTLSv1");
-  }
-#endif
-
-#ifdef SSL_OP_NO_DTLSv1_2
-  if (opts & SSL_OP_NO_DTLSv1_2) {
-    result.append(", SSL_OP_NO_DTLSv1_2");
-  }
-#endif
-
-#ifdef SSL_OP_NO_SSL_MASK
-  if (opts & SSL_OP_NO_SSL_MASK) {
-    result.append(", SSL_OP_NO_SSL_MASK");
-  }
-#endif
-
-#ifdef SSL_OP_PKCS1_CHECK_1
-  if (SSL_OP_PKCS1_CHECK_1) {
-    if (opts & SSL_OP_PKCS1_CHECK_1) {
-      result.append(", SSL_OP_PKCS1_CHECK_1");
-    }
-  }
-#endif
-
-#ifdef SSL_OP_PKCS1_CHECK_2
-  if (SSL_OP_PKCS1_CHECK_1) {
-    if (opts & SSL_OP_PKCS1_CHECK_2) {
-      result.append(", SSL_OP_PKCS1_CHECK_2");
-    }
-  }
-#endif
+void SslServerFeature::unprepare() {
+  LOG_TOPIC(TRACE, arangodb::Logger::SSL) << "unpreparing ssl: "
+                                          << stringifySslOptions(_sslOptions);
+  _context.reset();
+}
 
-#ifdef SSL_OP_NETSCAPE_CA_DN_BUG
-  if (SSL_OP_NETSCAPE_CA_DN_BUG) {
-    if (opts & SSL_OP_NETSCAPE_CA_DN_BUG) {
-      result.append(", SSL_OP_NETSCAPE_CA_DN_BUG");
-    }
+// called from GeneralServer::start
+void SslServerFeature::verifySslOptions() {
+  // check keyfile
+  if (_keyfile.empty()) {
+    LOG_TOPIC(FATAL, arangodb::Logger::SSL) << "no value specified for '--ssl.keyfile'";
+    FATAL_ERROR_EXIT();
   }
-#endif
 
-#ifdef SSL_OP_NETSCAPE_DEMO_CIPHER_CHANGE_BUG
-  if (opts & SSL_OP_NETSCAPE_DEMO_CIPHER_CHANGE_BUG) {
-    result.append(", SSL_OP_NETSCAPE_DEMO_CIPHER_CHANGE_BUG");
+  // validate protocol
+  if (_sslProtocol <= SSL_UNKNOWN || _sslProtocol >= SSL_LAST) {
```

Comment:
I think  "SSL_V2 = 1" is still mentioned in the options description. since its unsupported we should remove it.